internal/manager/config.go

```diff
@@ -185,16 +187,77 @@ func (c *Config) FlagSet() *pflag.FlagSet {
 	return flagSet
 }
 
+// CheckKongNonDefaultWorkSpace invokes :workspaces/ws-name, checks workspace existence
+func (c *Config) CheckKongNonDefaultWorkSpace(ctx context.Context, adminURL string, wsName string, client *kong.Client) (bool, error) {
+	exists, err := client.Workspaces.Exists(ctx, kong.String(wsName))
+	if err != nil {
+		return false, fmt.Errorf("failed looking up workspace: %w", err)
+	}
+
+	// if the provided workspace does not exist, for convenience we create it.
+	if !exists {
+		workspace := kong.Workspace{
+			Name: kong.String(wsName),
+		}
+		_, err := client.Workspaces.Create(ctx, &workspace)
+		if err != nil {
+			return false, fmt.Errorf("failed creating workspace: %w", err)
+		}
+		fmt.Printf("created workspace %s successfully", wsName)
+	}
+	return true, nil
+}
+
+// GetKongClient create a kong client based on configuration
 func (c *Config) GetKongClient(ctx context.Context) (*kong.Client, error) {
 	if c.KongAdminToken != "" {
+		fmt.Printf("kong-admin-token %s", c.KongAdminToken)
```

Comment:
we shouldn't print to stdout in library code. If we feel it's important to log this we should work the manager logger object into this function.

Reply:
https://github.com/Kong/kubernetes-ingress-controller/pull/1758/files#diff-f45c581008bda413c341fdf775a5220e0193e4d268db67eae49f6eabba06d729L191